website/docs/function_utils.md

```diff
@@ -0,0 +1,72 @@
+---
+id: function_utils
+title: Function Utilities
+sidebar_label: Function Utilities
+---
+
+## Function Utility Functions
+
+These functions provide a unified way to retrieve or format a function's parameters
+or get the calling module of a function.
+
+### Functions and Usage
+
+- `build_value_by_func_parameters` this function accepts a `key` parameter in type `str` which contains the parameter's placeholders and then returns out the string by replacing the placeholders with the parameter's actual value.
+
+    #### Usage
+```python
+from core_lib.helpers.func_utils import build_value_by_func_parameters
+
+def foo(param_1, param_2):
+    pass
+
+result = build_value_by_func_parameters('key_{param_1}_{param_2}', foo, 1, 2) # will return key_1_2
+result = build_value_by_func_parameters('key_{param_1}_{param_2}', foo, 1) # will return key_1_param_2
+```
+> **Note:** Will return the name of the placeholder if the parameter's value is missing.
+
+- `get_func_parameters_as_dict` will return the function's parameters formatted as a `dict` where key of the dictionary will be the parameter's name and value will be the value of the parameter.
+    

```

Comment:
please remove not needed words ``get_func_parameters_as_dict` will return the`
to ``get_func_parameters_as_dict` returns`
+ use grammerly please. 


try this 
``get_func_parameters_as_dict` extract a function parameters to dict.

```Python
def get_func_parameters_as_dict(func, *args, **kwargs) -> dict:
```
`func`: the function to extract all of it's parameters 
`*args, **kwargs`: the actual function args, kwargs